src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBinderFactory.cs

```diff
@@ -22,20 +24,42 @@ public class ModelBinderFactory : IModelBinderFactory
     {
         private readonly IModelMetadataProvider _metadataProvider;
         private readonly IModelBinderProvider[] _providers;
-
+        private readonly ILogger _logger;
         private readonly ConcurrentDictionary<Key, IModelBinder> _cache;
+        private readonly IServiceProvider _serviceProvider;
 
         /// <summary>
         /// Creates a new <see cref="ModelBinderFactory"/>.
         /// </summary>
         /// <param name="metadataProvider">The <see cref="IModelMetadataProvider"/>.</param>
         /// <param name="options">The <see cref="IOptions{TOptions}"/> for <see cref="MvcOptions"/>.</param>
+        [Obsolete("This constructor is obsolete and will be removed in a future version.")]
         public ModelBinderFactory(IModelMetadataProvider metadataProvider, IOptions<MvcOptions> options)
+            : this(metadataProvider, options, NullLoggerFactory.Instance, serviceProvider: null)
```

Comment:
This is going to cause failed `GetRequiredService<T>()` calls all over the place, Create a `static` method that does something like create a `ServiceCollection`, adds `NullLoggerFactory.Instance`, and builds the `IServiceProvider`. Then use that method here.